internal/entity/article.go

```diff
@@ -0,0 +1,119 @@
+package entity
+
+import (
+	"encoding/json"
+	"encoding/xml"
+)
+
+type ArticleList struct {
+	XMLName        xml.Name            `xml:"NewListInformation"`
+	ClubName       string              `xml:"ClubName"`
+	ClubWebsiteURL string              `xml:"ClubWebsiteURL"`
+	ArticleList    NewsletterNewsItems `xml:"NewsletterNewsItems"`
+}
+
+type NewsletterNewsItems struct {
+	XMLName  xml.Name              `xml:"NewsletterNewsItems"`
+	Articles []*NewsletterNewsItem `xml:"NewsletterNewsItem"`
+}
+
+type NewsletterNewsItem struct {
+	XMLName           xml.Name `xml:"NewsletterNewsItem"`
+	ArticleURL        string   `xml:"ArticleURL" bson:"articleUrl" json:"articleURL"`
+	NewsArticleID     int64    `xml:"NewsArticleID" bson:"_id" json:"newsArticleID"`
+	PublishDate       string   `xml:"PublishDate" bson:"publishDate" json:"publishDate"`
+	Taxonomies        string   `xml:"Taxonomies" bson:"taxonomies" json:"taxonomies"`
+	TeaserText        string   `xml:"TeaserText" bson:"teaserText" json:"teaserText"`
+	ThumbnailImageURL string   `xml:"ThumbnailImageURL" bson:"thumbnailImageURL" json:"thumbnailImageURL"`
+	Title             string   `xml:"Title" bson:"title" json:"title"`
+	OptaMatchId       string   `xml:"OptaMatchId" bson:"optaMatchId" json:"optaMatchId"`
+	LastUpdateDate    string   `xml:"LastUpdateDate" bson:"lastUpdateDate" json:"lastUpdateDate"`
+	IsPublished       string   `xml:"IsPublished" bson:"isPublished" json:"isPublished"`
+}
+
+func (a NewsletterNewsItem) MarshalJSON() ([]byte, error) {
```

Comment:
why do you need a custom MarshalJSON method here? 🤔

Reply:
Without method there are unnecessary fields like `XMLName` in response.
Method allows to prevent this behavior

Reply:
upd: fixed